extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/cache/NamespaceExtractionCacheManager.java

```diff
@@ -369,31 +356,28 @@ public void run()
               throw new NullPointerException(String.format("No data for namespace [%s]", id));
             }
             final Map<String, String> cache = getCacheMap(cacheId);
-            final String preVersion = implData.latestVersion.get();
+            final String preVersion = implData.latestVersion;
             final Callable<String> runnable = factory.getCachePopulator(id, namespace, preVersion, cache);
 
             tasksStarted.incrementAndGet();
             final String newVersion = runnable.call();
-            if (preVersion != null && preVersion.equals(newVersion)) {
-              throw new CancellationException(String.format("Version `%s` already exists", preVersion));
+            if (newVersion.equals(preVersion)) {
+              log.debug("Version `%s` already exists, skipping updating cache", preVersion);
+            } else {
+              updateNamespace(id, cacheId, newVersion);
+              log.debug("Namespace [%s] successfully updated", id);
             }
-            if (newVersion != null) {
-              if (!implData.latestVersion.compareAndSet(preVersion, newVersion)) {
-                log.wtf("Somehow multiple threads are updating the same implData for [%s]", id);
-              }
-            }
-            postRunnable.run();
-            log.debug("Namespace [%s] successfully updated", id);
           }
         }
         catch (Throwable t) {
-          delete(cacheId);
-          if (t instanceof CancellationException) {
-            log.debug(t, "Namespace [%s] cancelled", id);
```

Comment:
https://github.com/druid-io/druid/pull/3609/files#diff-52b91ace82cc677af64b49ff0b736baaR404 there is a check to make sure multiple things don't register with the same ID